crates/context/config/src/client/near.rs

```diff
@@ -0,0 +1,287 @@
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::time;
+
+pub use near_crypto::SecretKey;
+use near_crypto::{InMemorySigner, Signer};
+use near_jsonrpc_client::{methods, JsonRpcClient};
+use near_jsonrpc_primitives::types::query::{QueryResponseKind, RpcQueryResponse};
+use near_jsonrpc_primitives::types::transactions::{RpcTransactionError, TransactionInfo};
+use near_primitives::action::{Action, FunctionCallAction};
+use near_primitives::transaction::{Transaction, TransactionV0};
+pub use near_primitives::types::AccountId;
+use near_primitives::types::{BlockReference, FunctionArgs};
+use near_primitives::views::{
+    AccessKeyPermissionView, AccessKeyView, CallResult, FinalExecutionStatus, QueryRequest,
+    TxExecutionStatus,
+};
+use thiserror::Error;
+use url::Url;
+
+use super::{Operation, Transport, TransportRequest};
+
+#[derive(Debug)]
+pub struct NetworkConfig {
+    pub rpc_url: Url,
+    pub account_id: AccountId,
+    pub access_key: SecretKey,
+}
+
+#[derive(Debug)]
+pub struct NearConfig<'a> {
+    pub networks: BTreeMap<Cow<'a, str>, NetworkConfig>,
```

Comment:
Out of interest, what is the motivation for a copy-on-write string here? I am curious, because if it's a key it's unlikely to change? Perhaps we should make it a different type? Same with the other CoWs.

Note, this is an excellent example of where documentation would really help 😉

Reply:
`NearConfig` is a public struct, of which this `Cow` is a network ID, we don't keep this in the `NearTransport` that consumes the config, all we're saying is, give me a map between potential references to a string and a `NetworkConfig` in which case, different constructors could either have string slices or owned strings.

- `meroctl` will have owned strings, as a product of the config file deserialization
- the relayer, on the other hand, has static string slices